---
title: "formatieve opdracht metagenomics"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, echo=T, eval=F}
# update conda to the newest version and answer y on question proceed
echo "y" | conda update -n base conda

```

```{bash, engine.opts='-i', echo=T, eval=F}
# show all conda virtual environments
conda env list

# show yml file content
cat /home/berrit.kievith/daur2/metagenomics_species_identification/setup_meta_env.yml

```

```{bash, engine.opts='-i', echo=T, eval=F}
# create conda virtual environment
conda env create --file /home/berrit.kievith/daur2/metagenomics_species_identification/setup_meta_env.yml
```

```{bash, engine.opts='-i', echo=T, eval=F}
# activate environment
conda activate meta

# install fastqc and answer yes for procceed question
echo "y" | conda install -c bioconda fastqc

# deactivate conda virtual environment
conda deactivate
```

```{bash, engine.opts='-i', echo=T, eval=F}

# activate environment
conda activate meta

# perform FastQC analysis (change first path to your own folder structure if needed)
fastqc -o /home/berrit.kievith/daur2/metagenomics_species_identification/opdracht1/fastq/ /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R1_001.fastq.gz

fastqc -o /home/berrit.kievith/daur2/metagenomics_species_identification/opdracht1/fastq/ /home/daur2/metagenomics/formative_data/HU2_MOCK2_L001_R2_001.fastq.gz

# deactivate conda virtual environment
conda deactivate
```

```{r picture paste, echo=FALSE, fig.height=3, out.width= '100%', fig.cap="Fig. 1: Quality distribution: Right-side) Foward read quality scores range from 0 to 40 on the Phred scale. All bases for all reads have very high quality (>30), indicating that our reads have an accuracy of >99,9%. Left-side) reverse read quality scores ranging from 0 to 40 on the phred scale. the reads from 1 to 139 are very high quality from 140 to 151 the quality drops a little but the average read is still above 30." }

library(png)
library(grid)
library(gridExtra)

img1 <-  rasterGrob(as.raster(readPNG("opdracht1/fastq_pics/HU2_MOCK2_L001_R1_001.png")))
img2 <-  rasterGrob(as.raster(readPNG("opdracht1/fastq_pics/HU2_MOCK2_L001_R2_001.png")))
grid.arrange(img1, img2, ncol=2, top=textGrob("Per base quality distribution of forward (right) and reverse (left) reads", gp=gpar(fontsize=10,font=8)))


```







